File: OctopusNode/src/main/java/com/octopus/transport/encoders/TextEncoder.java
package com.octopus.transport.encoders;

import com.google.gson.Gson;

import javax.websocket.EncodeException;
import javax.websocket.Encoder;
import javax.websocket.EndpointConfig;

class TextEncoder<T> implements Encoder.Text<T> {
    private static Gson parser = new Gson();

    @Override
    public String encode(T object) throws EncodeException {
        return parser.toJson(object);
    }

    @Override
    public void init(EndpointConfig config) {}

    @Override
    public void destroy() {}
}

File: OctopusService/src/main/kotlin/com/octopus/transport/TextEncoder.kt
package com.octopus.transport

import com.fasterxml.jackson.databind.ObjectMapper
import javax.websocket.EncodeException
import javax.websocket.Encoder
import javax.websocket.EndpointConfig

open class TextEncoder<T> : Encoder.Text<T> {
    // Instantiated once per class
    companion object {
        private val mapper = ObjectMapper()
    }

    @Throws(EncodeException::class)
    override fun encode(obj: T): String {
        return mapper.writeValueAsString(obj)
    }

    override fun init(config: EndpointConfig) {}
    override fun destroy() {}
}